pkg/registry/generic/registry/store.go

```diff
@@ -40,6 +41,10 @@ import (
 	"github.com/golang/glog"
 )
 
+// EnableGarbageCollector affects the handling of Update and Delete requests. It
+// must be synced with the corresponding flag in kube-controller-manager.
+var EnableGarbageCollector bool
```

Comment:
Just wanted to point out that it's always safe to default the DeleteCollectionWorkers to 1, but it's not safe to always EnabledGarbageCollector to false, because that disables the orphaning ability. So it's more harmful if someone forgets propagate the `EnabledGarbageCollector`.

I think the solution would be create the Store via a initialization function rather than using struct literals, but that might hurt readability.

> Should maybe update to actually set the RESTOptions into the store 

I'm not sure I follow. Do you mean make RESTOptions a member of store?

> (or have a helper func responsible for setting DeleteCollectionWorkers, EnableGarbageCollection, etc, from the options)

But that still requires every NetREST() to call the helper func and someone can forget to call it.